src/passes/stringify-walker.h

```diff
@@ -65,6 +65,97 @@ struct StringifyWalker
 
   using Super = PostWalker<SubType, UnifiedExpressionVisitor<SubType>>;
 
+  struct SeparatorCtx {
+    struct FuncStart {
+      Function* func;
+    };
+
+    struct FuncEnd {
+      Function* func;
+    };
+
+    struct BlockStart {
+      Block* curr;
+    };
+
+    struct IfStart {
+      If* iff;
+    };
+
+    struct ElseStart {
+      If* iff;
+    };
+
+    struct LoopStart {
+      Loop* loop;
+    };
+
+    struct End {
+      Expression* curr;
+    };
+    using Separator = std::variant<FuncStart,
+                                   FuncEnd,
+                                   BlockStart,
+                                   IfStart,
+                                   ElseStart,
+                                   LoopStart,
+                                   End>;
+
+    Separator ctx;
+
+    SeparatorCtx(Separator ctx) : ctx(ctx) {}
```

Comment:
Maybe "Reason" instead of "Context" (i.e. Ctx) here?

Reply:
done